pallets/loans/src/lib.rs

```diff
@@ -629,6 +629,30 @@ pub mod pallet {
             Ok(().into())
         }
 
+        /// Sets a new liquidation incentive percentage for `currency_id`.
+        ///
+        /// Does nothing if the provided asset is not attached to an existent incentive.
+        ///
+        /// - `currency_id`: the asset that is going to be modified.
+        #[pallet::weight(T::WeightInfo::set_liquidation_incentive())]
+        #[transactional]
+        pub fn set_liquidation_incentive(
+            origin: OriginFor<T>,
+            currency_id: CurrencyId,
+            liquidate_incentive: Rate,
+        ) -> DispatchResultWithPostInfo {
+            T::UpdateOrigin::ensure_origin(origin)?;
+            Self::ensure_currency(&currency_id)?;
+
+            if <LiquidationIncentive<T>>::try_get(currency_id).is_err() {
+                return Err(<Error<T>>::CurrencyNotEnabled.into());
```

Comment:
into is not needed here.

Reply:
To convert `pallet::Error` into `DispatchErrorWithPostInfo`, it is necessary to call `into()` like in https://github.com/parallel-finance/parallel/blob/ee12e6e76e32ce36ecca6c76d57993fb92e9a081/pallets/loans/src/lib.rs#L1215